REST API implementation

zh opened the pull request:
Documentation: https://www.notion.so/Stackk-API-34e25e2a9c834bd6b864ca9c36aed2a6

In this PR:

[X] Initial REST API implementation
[X] Integration tests
[ X] Moderator requests signatures checking
[ X] User requests signatures checking

Tried first `bitcore-lib-cash` but sign/verify just does not work with BCH addresses. Because `Message` object is part of `bitcore-lib`, not from `bitcore-lib-cash` the methods inside are
checking the privite key to be instance of `bitcore-lib` private key. In case of signing, it still works with BCH private keys, because, as it is visible in the [examples](https://github.com/bitpay/bitcore/blob/master/packages/bitcore-lib-cash/docs/examples.md) they forcefully converting the key:

```js
var privateKey = new bitcore.PrivateKey(....) // see the type - bitcore 
```

But for verifying again a BCH address it just does not work - error message about wrong address network is returned. 

So I choose [bch-js](https://github.com/christroutner/bch-js) is used for signature checks. Why:

- BCH native, not bitcore-lib addition
- well supported - active development, a lot of examples
- async/await safe - created for usage with REST API - [bch-api](https://github.com/christroutner/bch-api)
- SLP addresses support

p0o commented on src/modules/comment/controller.js:
```diff
@@ -0,0 +1,158 @@
+const sanitizeHtml = require('sanitize-html')
+const Comment = require('../../models/comment')
+
+let _this
```
This is a old OOP pattern used mostly as workarounds for specific reasons. Is there any reason we need to use it here?

p0o commented on util/create_wallet.js:
```diff
@@ -0,0 +1,37 @@
+
+const COUNT = 10
+
+async function generateTestWallet (addressesCount) {
```
I appreciate the work but I think we don't really need extra libraries to build wallets for testing signatures. Hard coded values of signature and texts takes less time for tests to run and are immutable without exposing anything critical.

zh commented:
In commit# df812f9

* `bch-js` removed - signature verification by `bitcore-message` (no `sign()` method provided)
* fixed signatures = faster tests, but changes will be difficult - signatures are just a text now
* no more `create-wallet.js` - there is `test/wallet.json` file with 10 example addresses
* About cleaning DB - without it, tests complain about duplicated records IDs. As a safety measure, added a check and test **will not run unless env=test**

p0o reviewed (approved):
Thanks for the great work.
